it/controllers/employment/EmployerPayAmountControllerISpec.scala

```diff
@@ -0,0 +1,341 @@
+          formPostLinkCheck(continueButtonLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "should render How much did xxx pay you? page with generic paragraph text when user is adding a new employment" which {
+
+          implicit lazy val result: WSResponse = {
+            authoriseAgentOrIndividual(user.isAgent)
+            dropEmploymentDB()
+            val newCya = CyaModel.cya.copy(employment = CyaModel.cya.employment.copy
+            (employmentDetails = CyaModel.cya.employment.employmentDetails.copy(taxablePayToDate = None)))
+            insertCyaData(newCya, User(mtditid, None, nino, sessionId, "agent"))
+            urlGet(urlEOY, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          titleCheck(get.expectedTitle)
+          h1Check(get.expectedH1)
+          captionCheck(expectedCaption(taxYearEOY))
+          textOnPageCheck(hintText, hintTestSelector)
+          textOnPageCheck(get.expectedContentNewAccount, contentSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldCheck(amountInputName, inputSelector)
+          inputFieldValueCheck("", inputAmountField)
+
+          buttonCheck(continueButtonText, continueButtonSelector)
+          formPostLinkCheck(continueButtonLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+
+        }
+      }
+    }
+
+  }
+
+  ".show" should {
+
+    "redirect  to check employment details page when there is no cya data in session" when {
+      implicit lazy val result: WSResponse = {
+        authoriseAgentOrIndividual(isAgent = false)
+        dropEmploymentDB()
+        urlGet(urlEOY, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+      }
+
+
+      "has an SEE_OTHER status" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some("/income-through-software/return/employment-income/2021/check-employment-details?employmentId=001")
+      }
+    }
+
+    "redirect  to overview page if the user tries to hit this page with current taxYear" when {
+      implicit lazy val result: WSResponse = {
+        authoriseAgentOrIndividual(isAgent = false)
+        dropEmploymentDB()
+        insertCyaData(CyaModel.cya, User(mtditid, None, nino, sessionId, "agent"))
+        val inYearUrl =s"$appUrl/$taxYear/how-much-pay?employmentId=001"
+        urlGet(inYearUrl, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+      }
+
+
+      "has an SEE_OTHER status" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some("http://localhost:11111/income-through-software/return/2022/view")
+      }
+    }
```

Comment:
these show tests could be moved up to the prior .show tests and utilise the us.isAgent to cover more scenarios when they're an agent